src/Debugger-Model/DebugSession.class.st

```diff
@@ -265,16 +265,18 @@ DebugSession >> name: aString [
 
 { #category : #context }
 DebugSession >> pcRangeForContext: aContext [
+
 	"Answer the indices in the source code for the method corresponding to 
 	aContext's program counter value."
-	(aContext isNil) ifTrue: [ 
-		self terminate.
-		 ] ifFalse: [  
-	(aContext isDead) ifTrue:
-		[^1 to: 0].
-	^aContext debuggerMap
-		rangeForPC: aContext pc
-		contextIsActiveContext: (self isLatestContext: aContext) ]
+
+	aContext
+		ifNil: [ self terminate ]
```

Comment:
Nice how this refactoring shows a bug: in the ifNil: case, the method returns a DebugSession, not a range

Reply:
(we should merge this and add a new issue. Rule 1 of tiny changes: *only* do the tiny change, everything else should be added to the backlog)

Reply:
added #3979